src/Jackalope/Transport/DoctrineDBAL/Client.php

```diff
@@ -823,9 +823,7 @@ private function syncNode($uuid, $path, $type, $isNewNode, $props = array(), $pr
             $this->syncBinaryData($nodeId, $propsData['binaryData']);
         }
 
-        if (!empty($propsData['references'])) {
-            $this->referencesToUpdate[$nodeId] = array('path' => $path, 'properties' => $propsData['references']);
-        }
+        $this->referencesToUpdate[$nodeId] = array('path' => $path, 'properties' => $propsData['references']);
```

Comment:
i opt for prudence over efficiency. in the end there is no way to know if there previously where references from this node to another. properties can be deleted or the type can change or values removed from multivalue properties...